Services/Tracking/classes/status/class.ilLPStatusCollection.php

```diff
@@ -323,6 +329,44 @@ public static function determineGroupingStatus(
         return $status;
     }
 
+    /**
+     * @param int $objId
+     * @param int $usrId
+     * @return bool
+     */
+    protected function isMember(int $objId, int $usrId) : bool
+    {
+        global $DIC;
+
+        $ilObjDataCache = $DIC['ilObjDataCache'];
+        $tree = $DIC['tree'];
+
+        switch ($ilObjDataCache->lookupType($objId)) {
+            case 'crs':
+                $participants = ilCourseParticipant::_getInstanceByObjId($objId, $usrId);
+                return $participants->isMember();
+
+            case 'grp':
+                $participants = ilGroupParticipants::_getInstanceByObjId($objId);
+                return $participants->isMember($usrId);
+
+            case 'fold':
+                $folderRefIds = ilObject::_getAllReferences($objId);
+                $folderRefId = current($folderRefIds);
+                if ($crsRefId = $tree->checkForParentType($folderRefId, 'crs')) {
```

Comment:
Please implement an additional check for group membership here: checkForParentType($foldeRefId, 'grp');